src/go/k8s/internal/controller/redpanda/cluster_controller.go

```diff
@@ -1023,9 +985,43 @@ func (r *ClusterReconciler) doDecommissionGhostBrokers(c context.Context, vClust
 	if err != nil {
 		return fmt.Errorf("unable to create admin client: %w", err)
 	}
-	adminBrokers, err := adminClient.Brokers(ctx)
+
+	health, err := adminClient.GetHealthOverview(ctx)
+	if err != nil {
+		return fmt.Errorf("feating cluster heatlh overview: %w", err)
+	}
+
+	if !health.IsHealthy {
+		err = errors.New("cluster is not healthy")
+		log.V(logger.DebugLevel).Error(err, "stopping decommission verification",
+			"requested-replicas", vCluster.Spec.Replicas,
+			"nodes-registered", health.AllNodes,
+			"nodes-down", health.NodesDown)
+		return err
+	}
+
+	ss, err := ar.getStatefulSet()
 	if err != nil {
-		return fmt.Errorf("unable to fetch brokers: %w", err)
+		return err
+	}
+
+	sts := &appsv1.StatefulSet{}
+	if err := r.Client.Get(ctx, ss.Key(), sts); err != nil {
+		return fmt.Errorf("could not retrieve the statefulset: %w", err)
+	}
+
+	if vCluster.Spec.Replicas == nil {
+		return nil
+	}
+
+	if len(health.AllNodes) <= int(*vCluster.Spec.Replicas) {
+		log.V(logger.DebugLevel).Info("not enough nodes to check decommission", "all-nodes", health.AllNodes, "replicas", *vCluster.Spec.Replicas)
+		return nil
+	}
+
+	if sts.Status.Replicas != *vCluster.Spec.Replicas || sts.Status.AvailableReplicas != *vCluster.Spec.Replicas {
```

Comment:
Technically speaking wouldn't this check imply the one above or visa versa?

Reply:
I agree. I always tries to have confirmation from multiple signals. One would come from Redpanda itself and second would come from Kuberntes perspective